src/linker/Linker/ConsoleLogger.cs

```diff
@@ -3,14 +3,41 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Collections.Generic;
 
 namespace Mono.Linker
 {
 	public class ConsoleLogger : ILogger
 	{
-		public void LogMessage (MessageContainer message)
+		readonly List<MessageContainer> _messageContainers;
```

Comment:
Any reason why this is tied to the ConsoleLogger? I would prefer if we have the same behavior across all loggers. Especially since there's no technical reason anymore. Basically build this into the context and just call Logger.LogMessage after sorting to output.